msteams-platform/toolkit/debug-message-extension-app-in-test-tool.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Debug message extension app in Test Tool
+author: surbhigupta 
+description: Learn how to emulate the Teams experience for your message extension app in Teams App Test Tool.
+ms.author: surbhigupta 
+ms.localizationpriority: high
+ms.topic: overview
+ms.date: 04/25/2024
+---
+
+# Debug message extension app in Test Tool
+
+You can debug bot-based message extension capabilities such as search commands, action commands, and link unfurling using Teams App Test Tool (Test Tool) within a simulated environment. Test tool helps you to identify and resolve issues before deploying the extension to a live environment.
+
+Test Tool enhances the functionality of bot-based message extensions, which are built on top of the Bot Framework. When a message extension is activated, the Test Tool sends an invoke request to the app. The app then processes this request and returns an invoke response, which the Test Tool renders and displays.
+
+> [!NOTE]
+> The user experience to trigger message extension in Test Tool is different from Teams, as the goal of Test Tool is to test and debug the app logic and make the flow simple, instead of being a real chat app.
+
+## Prerequisites
+
+Ensure that you install the following tools for building and deploying your bots in Test Tool:
+
+| &nbsp; | Install | For using... |
+| --- | --- | --- |
+| &nbsp; | [Teams Toolkit](install-Teams-Toolkit.md) | A Microsoft Visual Studio Code extension that creates a project scaffolding for your app. Use the latest prerelease version. |
+| &nbsp; | [Node.js](https://nodejs.org/en/download/) | Back-end JavaScript runtime environment. For more information, see [Node.js version compatibility table for project type](~/toolkit/build-environments.md#nodejs-version-compatibility-table-for-project-type). |
+| &nbsp; | [Visual Studio Code](https://code.visualstudio.com/download) | JavaScript, TypeScript, or SharePoint Framework (SPFx) build environments. Use the latest version. |
+
+## Build and run the sample app
```

Comment:
As the document of [Debug bot using Teams App Test Tool](https://learn.microsoft.com/en-us/microsoftteams/platform/toolkit/debug-your-teams-app-test-tool?tabs=vscode%2Cclijs) go through the bot app template scaffolded by Teams Toolkit, shall we also use the ME app template scaffolded by Teams Toolkit in this document to **keep consistent**?

Reply:
Message extension is used.